plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsConnectorTableHandle.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.google.sheets;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.NotFoundException;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.TableNotFoundException;
+
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        property = "@type")
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = SheetsNamedTableHandle.class, name = "named"),
+        @JsonSubTypes.Type(value = SheetsSheetTableHandle.class, name = "sheet")
+})
+public sealed interface SheetsConnectorTableHandle
+        extends ConnectorTableHandle
+        permits SheetsNamedTableHandle, SheetsSheetTableHandle
+{
+    default NotFoundException tableNotFound()
```

Comment:
Putting this method in the interface looks weird to me.

Reply:
Where would you like the method to exist instead? Making it a static method in a utility class seems worse.

Reply:
Probably, we can change `Optional<SheetsTable> SheetsClient.getTable(SheetsConnectorTableHandle tableHandle)`  to non-optional and move this method to around there. If we still want to use an interface, moving the implementation to `SheetsNamedTableHandle` to `SheetsSheetTableHandle` looks better to me.

Reply:
Moved the implementation to the implementing classes. Removing the optional is more difficult since `SheetsMetadata#getTableHandle` returns null when the optional is empty while other methods throw.